core/metadata.js

```diff
@@ -360,10 +360,14 @@ const makeComparator = (name, interestingFields) => {
       })
     })
   }
+  const isRealDiff = (difference) => {
+    const diff = difference.item ? difference.item : difference
+    return _.isNil(diff.lhs) && _.isNil(diff.rhs)
+  }
   return (one, two) => {
     const diff = deepDiff(one, two, filter)
     log.trace({path: two.path, diff}, name)
-    return !diff
+    return !diff || _.every(diff, isRealDiff)
```

Comment:
I wonder whether replacing sameFile/sameFolder with two fileDiff/folderDiff functions returning either a diff object or nothing would make both the producer/consumer code simpler...

Although I wouldn't be comfortable doing this without more test coverage on Merge.

(also I wonder whether using deep-diff was such a good idea, my bad in case it wasn't, but this one is a smaller issue, out of the scope of this PR)

Reply:
I feel like this is pretty much what we're doing already except we return a boolean instead of the diff itself.
As for using deep-diff, I think it's a nice tool even if some of their choices don't match with our needs. Those differences are quite small imho and don't justify moving away from it.

Reply:
:ok_hand: out of the scope of this PR